hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/support/validators/block/StateChangesValidator.java

```diff
@@ -231,7 +232,7 @@ public StateChangesValidator(
                         metrics),
                 DiskStartupNetworks::new,
                 NodeId.of(0L));
-        this.state = (MerkleStateRoot) hedera.newMerkleStateRoot();
+        this.state = (PlatformMerkleStateRoot) hedera.newMerkleStateRoot();
```

Comment:
I'm confused why there are so many changes to the `MerkleStateRoot` interface. Doing so requires changes like we see here, where we go from an interface to something _more_ specific, despite this not being a platform test of any kind. What is the advantage of these changes?

Reply:
The fundamental problem with the current `MerkleStateRoot` implementation is that it plays two roles:
- a gateway between the app and the platform
- Merkle-tree-based implementation of the State API

This PR partially addresses that - it separates platform-related code from the state implementation. Note that there is a follow-up ticket to refactor `swirlds-platform-core` further and get rid of `PlatformMerkleStateRoot`
https://github.com/hashgraph/hedera-services/issues/16568. 

Once this PR is merged and this , Block Node will be able to use `swirlds-state-api` and `swirlds-state-impl` to work with the state, and there will be no need to depend on `swirlds-platform-core`. This PR is a part of a broader effort - https://github.com/hashgraph/hedera-services/issues/15574 - which has it as the end goal.